Assets/HoloToolkit/Input/Scripts/Microphone/MicStream.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using UnityEngine;
+using System.Text;
+
+namespace HoloToolkit.Unity
+{
+	public class MicStream
+	{
+		// This class replaces Unity's Microphone object
+		// This class is made for HoloLens mic stream selection, but should work well on all windows 10 devices
+		// chooses from one of three possible microphone modes on HoloLens
+
+		// Streams: SPEECH is optimized for voice transmission, COMMUNICATIONS is higher quality voice, MEDIA is a "room capture"
+		// can only be set on initialization
+		public enum StreamCategory { SPEECH, COMMUNICATIONS, MEDIA }
+
+		public enum ErrorCodes { ALREADY_RUNNING = -10, NO_AUDIO_DEVICE, NO_INPUT_DEVICE, ALREADY_RECORDING, GRAPH_NOT_EXIST, CHANNEL_COUNT_MISMATCH, FILE_CREATION_PERMISSION_ERROR, NOT_ENOUGH_DATA };
+		
+		[UnmanagedFunctionPointer(CallingConvention.StdCall)] // If included in MicStartStream, this callback will be triggered when audio data is ready. This is not the preferred method for Game Engines and can probably be ignored.
+		public delegate void LiveMicCallback();
+
+		/// <summary>
+		/// Called before calling MicStartStream or MicstartRecording to initialize microphone
+		/// </summary>
+		/// <param name="category">One of the entries in the StreamCategory enumeratio</param>
+		/// <returns>error code or 0</returns>
+		[DllImport("MicStreamSelector", ExactSpelling = true)]    public static extern int MicInitializeDefault(int categoryh);
```

Comment:
NIT: These DllImports might be more readable with the function declaration being in a new line. For example:
[DllImport(....)]
public static void FuncName( )